src/modules/paymentProcessor/PP_Streaming_v1.sol

```diff
@@ -948,23 +957,48 @@ contract PP_Streaming_v1 is Module_v1, IPP_Streaming_v1 {
         );
     }
 
-    function _getStreamingDetails(bytes32 flags, bytes32[] memory data)
-        internal
-        view
-        returns (uint start, uint cliff, uint end)
-    {
-        uint dataIdx = 0;
+    function _getStreamingDetails(
+        bytes32 order_flags,
+        bytes32[] memory order_data
+    ) internal view returns (uint start_, uint cliff_, uint end_) {
+        uint[3] memory return_data;
+
+        uint8 position_in_order_data = 0;
+        uint8 position_in_return_data = 0;
 
-        bool hasStart = (uint(flags) & (1 << 1)) != 0;
-        start = hasStart ? uint(data[dataIdx]) : defaultStart;
-        if (hasStart) dataIdx += 1;
+        for (uint i = 0; i < 256; i++) {
+            if (position_in_return_data == return_data.length) {
+                // we have either:
+                // - reached the end of the order_data array
+                // - already checked for all the values this PaymentProcessor will need
+                //      ==> exit loop
+                break;
+            }
 
-        bool hasCliff = (uint(flags) & (1 << 2)) != 0;
-        cliff = hasCliff ? uint(data[dataIdx]) : defaultCliff;
-        if (hasCliff) dataIdx += 1;
+            //uint order_bit = (uint(order_flags) & (1 << i));
```

Comment:
Missing whitespaces between `//` and the comment.
What are these comments as well? What do these do here?